.github/workflows/ci.snyk.yml

```diff
@@ -0,0 +1,23 @@
+name: Snyk
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'develop'
+
+jobs:
+  test:
+    # Only run on push events or PRs from iotaledger/firefly, skip on PRs from forks
+    # Secret variables cannot be exposed to PRs from forks
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name == github.repository
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Check vulnerabilities
+        uses: snyk/actions/node@0.3.0
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        with:
+          args: --yarn-workspaces --strict-out-of-sync=false --detection-depth=5 --dev
```

Comment:
Per https://docs.snyk.io/products/snyk-open-source/language-and-package-manager-support/snyk-for-javascript setting `--strict-out-of-sync=false` will
> Prevent testing out-of-sync lockfiles (test fails when set to true if there are out-of-sync lockfiles in the project)

That seems like it would cause inaccurate results. For example, if the lock file specified a vulnerable dependency version (say `a==1.0`), but the `package.json` file specified a range (ex `^1.0.0`), then with this argument snyk would resolve the dependency to a later version than is really used by the application (in this case, `a==1.1` which doesn't have the vulnerability).

Reply:
Yep, this flag isn't ideal but it seems necessary for Yarn workspaces due to a bug that's still open (https://github.com/snyk/cli/issues/350#issuecomment-762956088). We enforce that the lockfile is updated during code review and are looking to add this to our CI pipeline though.

Reply:
Perhaps you could submit a minimal reproducible example to the Snyk issue, so they can easily reproduce this issue and fix it?

Reply:
Two projects were provided by someone in https://github.com/snyk/cli/issues/350#issuecomment-854246766 and it seems like those were forwarded to the appropriate team. I unfortunately have not had the bandwidth to create a minimal reproducible example yet.